workflow-runtime/src/main/java/com/squareup/workflow1/internal/SubtreeManager.kt

```diff
@@ -110,8 +110,10 @@ internal class SubtreeManager<PropsT, StateT, OutputT>(
     // and must be torn down.
     children.commitStaging { child ->
       child.workflowNode.cancel()
-      snapshotCache -= child.id
```

Comment:
I did some blame digging, and this line has been here for a few commits. Doesn't make sense to me, so yea this change definitely seems like a big improvement.